athenet/sparsifying/utils/interval.py

```diff
@@ -0,0 +1,364 @@
+"""Intervals in Theano including special functions for sparsifying."""
+
+from athenet.sparsifying.utils.numlike import Numlike
+from theano import function
+from theano import tensor as T
+from theano.ifelse import ifelse
+
+
+class Interval(Numlike):
+    """Theano interval matrix class
+
+    represents matrix of interval. Behaves like limited numpy.ndarray.
+
+    .. note:: Should be treated as interval type with bounds as Theano nodes.
+              Opetations on Interval create nodes in Theano graph. In order to
+              read result of given operations, use eval method.
+    """
+
+    def __init__(self, lower, upper):
+        """Creates interval.
+
+        :param lower: lower bound of Interval to be set
+        :param upper: upper bound of Interval to be set
+        :type lower: numpy.ndarray or theano variable
+        :type upper: numpy.ndarray or theano variable
+
+        .. note:: lower must be lower than upper. It is not being checked.
```

Comment:
[https://imgflip.com/i/11gy72](url)

1. Safety
2. Efficiency

You'll remove it when you'll end the algorithm and you will be sure that works properly